002_simulation_structure/002_simulation_structure.tex

```diff
@@ -100,6 +100,210 @@ \subsection{Infrastructure}
+    \item \textbf{\texttt{LeavePercentFood}}: Requests a signed agent to leave a percentage of food on the platform.
+    \item \textbf{\texttt{Inform}}: Requests a signed agent to alert their neighbor if treaty conditions are met.
+\end{itemize} \\ 
+ \hline
+ \texttt{RequestOp} & Includes all the mathematical operators $\geq, >, =, <, \leq$ \\ 
+ \hline
+ \texttt{RequestValue} & Value that the \texttt{RequestType} has to meet the \texttt{RequestOp} for. \\ 
+ \hline
+\end{tabular}
+\end{center}
+\vspace{\baselineskip}
+The aforementioned treaty \textit{“Please leave 40 food on the platform if you are not in critical condition”} would be translated into \newline
+\texttt{ConditionType = HP \newline
+        ConditionOp = GT \newline
+        ConditionValue = 20 //(where 20 is the critical threshold) \newline
+        RequestType = LeaveAmountFood \newline
+        RequestOp = EQ \newline
+        RequestValue = 40 \newline}
+In addition to the language components the treaties included a \texttt{SignatureCount} which tracked an estimate of how many agents had signed the treaty thus far. This number was not an accurate, it only vaguely indicates whether or not a treaty is popular. 
+The introduction of treaties also necessitated the creation of one more message category as well as a message sub-type. 
+\begin{center}
+\begin{tabular}{p{3cm}p{3cm}p{5cm}p{3.5cm}}
+ \hline
+ \textbf{Message \newline Category} & \textbf{Reply \newline Category} & \textbf{Body Functions} & \textbf{Description} \\ [0.5ex] 
+ \hline\hline
+ \texttt{ProposalMessage} & \texttt{ResponseMessage} &  
+ \texttt{Reply}: Returns \texttt{TreatyResponseMessage} \newline 
+ \texttt{Treaty}: Returns the treaty that the proposal is carrying &
+ Carries a treaty from proposer/propagator to recipient.\\
+ \hline
+\end{tabular}
+\end{center}
+\texttt{TreatyResponse} sub-type of \texttt{ResponseMessage} which were response messages that contained the ID of the treaty it was responding to. \newline
+The treaties-relay-design and the inaccuracy of the signature count is illustrated if Alice, still on Floor 10, wanted to propose the treaty \textit{“Please leave 40 food on the platform if you are not in critical condition”} to her neighbours downstairs and upstairs it would work as follows:
+\begin{enumerate}
+    \item Tick 0: Alice constructs the treaty, embeds it within a \texttt{ProposeTreatyMessage} and sends it upward (as she can only send one message per tick). Alice's treaty begins with a \texttt{SignatureCount} of 1 (since she implicitly signed it by proposing).
+    \item Tick 1: Treaty arrives on the 11th Floor in Bob's (previously empty) inbox and he chooses to respond positively and signs. On Bob's local copy of the treaty, the \texttt{SignatureCount} has incremented to 2 but not on Alice's. Bob replies through \texttt{Reply} and sends the  \texttt{TreatyResponse} downstairs.
+    \item Tick 1: Meanwhile, Alice also sends the treaty downward to Floor 9.
```

Comment:
```suggestion
\texttt{TreatyResponse} sub-type of \texttt{ResponseMessage} which were response messages that contained the ID of the treaty it was responding to. \newline
The treaties-relay-design and the inaccuracy of the signature count is illustrated if Alice, still on Floor 10, wanted to propose the treaty \textit{“Please leave 40 food on the platform if you are not in critical condition”} to her neighbours downstairs and upstairs it would work as follows:
\begin{enumerate}
    \item Tick 0: Alice constructs the treaty, embeds it within a \texttt{ProposeTreatyMessage} and sends it upward (as she can only send one message per tick). Alice's treaty begins with a \texttt{SignatureCount} of 1 (since she implicitly signed it by proposing).
    \item Tick 1: Treaty arrives on the 9th Floor in Bob's (previously empty) inbox and he chooses to respond positively and signs. On Bob's local copy of the treaty, the \texttt{SignatureCount} has incremented to 2 but not on Alice's. Bob replies through \texttt{Reply} and sends the  \texttt{TreatyResponse} downstairs.
    \item Tick 1: Meanwhile, Alice also sends the treaty downward to Floor 11.
```